src/communication/bolt/v1/encoder/client_encoder.hpp

```diff
@@ -38,22 +44,20 @@ class ClientEncoder : private BaseEncoder<Buffer> {
   /**
    * Writes a Init message.
    *
-   * From the Bolt v1 documentation:
-   *   InitMessage (signature=0x01) {
-   *     String clientName
-   *     Map<String,Value> authToken
+   * From the Bolt v4.3 documentation:
+   *   HelloMess (signature=0x01) {
+   *     Map<String,Value> extra
    *   }
    *
    * @param client_name the name of the connected client
    * @param auth_token a map with authentication data
    * @returns true if the data was successfully sent to the client
    *          when flushing, false otherwise
    */
-  bool MessageInit(const std::string client_name, const std::map<std::string, Value> &auth_token) {
-    WriteRAW(utils::UnderlyingCast(Marker::TinyStruct2));
+  bool MessageInit(const std::map<std::string, Value> &extra) {
+    WriteRAW(utils::UnderlyingCast(Marker::TinyStruct1));
     WriteRAW(utils::UnderlyingCast(Signature::Init));
-    WriteString(client_name);
-    WriteMap(auth_token);
+    WriteMap(extra);
```

Comment:
If you are changing how we handle messages to fit v3 bolt version, does this mean we will be incompatible with v1? Have you tested these changes so that clients using v1 are not affected?

Reply:
This `ClientEncoder` is only meant to support one specific Bolt version. Previously it supported only Bolt v1 and I upgraded this to Bolt v4.3. So you are right, this client will be incompatible with Bolt v1, but Memgraph will work well with Bolt v1, as these changes only affect the C++ client and not the Bolt server we use in Memgraph.